rust_src/src/fonts.rs

```diff
@@ -0,0 +1,71 @@
+use remacs_macros::lisp_fn;
+use remacs_sys::{font, Qfont_spec, Qfont_entity, Qfont_object, wrong_type_argument};
+use lisp::LispObject;
+use symbols::intern;
+use vectors::LispVectorlikeRef;
+
+// A font is not a type in and of itself, it's just a group of three kinds of
+// pseudovector. This newtype allows us to define methods that yield the actual
```

Comment:
I'm glad someone is reading the docstrings :) Not quite sure I agree about the plural -- a font is "one of three kinds of pseudovector", so fonts are "a group of three kinds of pseudovector". Might be wrong here, though.

The term "newtype" [refers to](https://aturon.github.io/features/types/newtype.html) the Rust pattern of `struct X(Y)`. It's not a typo. A "font" does not have its own structure in the C code. It refers to three possible structs. Hence a font is not "a type in and of itself". It's just a label for a group of structures. So I'm not actually defining any structure here, I just want to add some methods to `LispVectorlikeRef` -- hence the newtype.